test/e2e-tests.sh

```diff
@@ -60,8 +60,10 @@ go_test_e2e -timeout=10m \
   ./test/scale || failed=1
 
 # Auto TLS E2E tests mutate the cluster and must be ran separately
+kubectl apply -f ./test/config/autotls/certmanager/selfsigned/
 go_test_e2e -timeout=10m \
   ./test/e2e/autotls || failed=1
+kubectl delete -f ./test/config/autotls/certmanager/selfsigned/
```

Comment:
do you need to trap this? in case I kill the test with sigkill?

Reply:
I added an extra line with `trap` to handle sigkill. I still keep the original line as we want to clean up the cert-manager related resources after the auto tls test finishes in order to not interfere the rest of tests (although it is not likely the rest of the tests would be affected even with those cert-manager related configs).

Reply:
trap should be added above kubectl apply I believe. Also, `trap` does override other traps I believe, please combine if there is any trap with same signal somewhere else

Reply:
fixed this by adding `add_trap` function.